gravitee-am-repository/gravitee-am-repository-mongodb/src/main/java/io/gravitee/am/repository/mongodb/management/AbstractUserRepository.java

```diff
@@ -224,8 +227,8 @@ public Flowable<User> findByIdIn(List<String> ids) {
     }
 
     @Override
-    public Maybe<User> findById(ReferenceType referenceType, String referenceId, String userId) {
-        return Observable.fromPublisher(usersCollection.find(and(eq(FIELD_REFERENCE_TYPE, referenceType.name()), eq(FIELD_REFERENCE_ID, referenceId), eq(FIELD_ID, userId))).first()).firstElement().map(this::convert);
+    public Maybe<User> findById(Reference reference, UserId userId) {
+        return Observable.fromPublisher(usersCollection.find(and(eq(FIELD_REFERENCE_TYPE, reference.type().name()), eq(FIELD_REFERENCE_ID, reference.id()), userIdMatches(userId))).first()).firstElement().map(this::convert);
```

Comment:
**question**: I wondering if it is safe to use userIdMatch in the UserRepositories :thinking: I would prefer to stay with a search by id to avoid side effect...